packages/editor-ui/src/components/RunData.vue

```diff
@@ -727,8 +727,9 @@ export default mixins(
 				localStorage.setItem(LOCAL_STORAGE_PIN_DATA_DISCOVERY_CANVAS_FLAG, 'true');
 			},
 			enterEditMode({ origin }: EnterEditModeArgs) {
-				const data = this.rawInputData && this.rawInputData.length > 0
-					? this.convertToJson(this.rawInputData)
+				const inputData = this.pinData ? this.pinData as PinData : this.convertToJson(this.rawInputData);
```

Comment:
`as PinData` → We should type this properly in the mixin.

Reply:
Updated!